### ISSUE TITLE
 No TimeSpan support in Schema.Parse()

### ISSUE THREAD

**Issue by JaseRandall** [MarkPflug/Sylvan#257]
Hi @MarkPflug

There is no TimeSpan support in Schema.Parse(). Is that something that can be changed?

"Duration:TimeSpan"
System.ArgumentException: 'Value does not fall within the expected range.'

**Comment by MarkPflug** [comment]
Yeah, the schema serialization code is... not great, and could certainly be improved. I just haven't personally had any need for it. Do you actually need to persist the schema as a string, or can you use the builder API instead?

```C#
static Schema GetSchema()
{
	return
		new Schema.Builder()
		.Add<int>("Id")
		.Add<string>("Description")
		.Add<DateTime>("StartTime")
		.Add<TimeSpan>("Duration")
		.Build();
}
```
The `Schema` type itself isn't restricted in the types it can express, only the serializer code behind the `Parse` method.

**Comment by JaseRandall** [comment]
Thanks for the suggestion. I'm unable to use that `Schema.Builder()` implementation as I persist the schema in a DB then at runtime determine which schema applies to the file then build the string to pass into `Schema.Parse()`.

However, in having another look at how it all fits together I saw that I could pass in an `IEnumerable<DbColumn>` to the constructor of  `CsvSchema`. I changed my "column details" class to inherit `DBColumn` and then I basically had an `IEnumerable<DbColumn>` ready to use. So, I've gone with that approach instead and now don't need to use `Schema.Builder()`.
